### DISCUSSION TITLE
One Class Segmentation Model Ignoring Background Class

### DISCUSSION THREAD

**Discussion by ghost** [azavea/raster-vision#2247] [General]
Just wondering, if it is possible to train one class model, ignoring background. I noticed, sometime 1-class segmentation model works pretty well, however, when we add background to a class, it reduce the accuracy. 

Thanks

**Comment by AdeelH** [comment]
No, Raster Vision does not have built-in support for training on a single binary class with BCE loss. It uses cross-entropy loss by default. 

You might still be able to do that by setting
```py
# this stops RV from adding a separate null class, but note that this will cause
# NODATA pixels to be assigned the my_class class (if doing semantic segmentation),
# so use with caution
class_config = ClassConfig(names=['my_class'], null_class='my_class')
```
and by specifying an external loss like shown here: 
https://github.com/azavea/raster-vision/blob/v0.31.1/rastervision_pytorch_backend/rastervision/pytorch_backend/examples/chip_classification/spacenet_rio.py#L162-L180

and for that you will need to have a repo or directory with a `hubconf.py` that contains that loss function.

**Comment by ghost** [comment]
Great thanks for the reply. And thank you for building awesome open source library. Much appreciated!
